tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -675,26 +700,23 @@ func (s *storeImpl) Walk(ctx context.Context, fn func(obj *{{.Type}}) error) err
 	return nil
 }
 
-{{ if eq .ResourceType "directlyScoped" }}
-    func isInScope(obj *{{.Type}}, eas *effectiveaccessscope.ScopeTree) bool {
+{{ if .Obj.IsDirectlyScoped }}
+func isInScope(obj *{{.Type}}, eas *effectiveaccessscope.ScopeTree) bool {
     if eas.State == effectiveaccessscope.Included {
         return true
     }
-    if eas.State == effectiveaccessscope.Excluded {
-        return false
-    }
-    clusterId := {{ .ClusterGetter }}
+    clusterId := {{ "obj" | .Obj.GetClusterID }}
     cluster := eas.GetClusterByID(clusterId)
-    {{ if not .NamespaceGetter -}}
-    return cluster.State == effectiveaccessscope.Included
+    {{ if .Obj.IsClusterScope -}}
+    return cluster != nil && cluster.State == effectiveaccessscope.Included
     {{  else -}}
+    if cluster == nil || cluster.State == effectiveaccessscope.Excluded {
+        return false
+    }
     if cluster.State == effectiveaccessscope.Included {
         return true
     }
-    if cluster.State == effectiveaccessscope.Excluded {
-        return false
-    }
-    namespaceName := {{ .NamespaceGetter }}
+    namespaceName := {{ "obj" | .Obj.GetNamespace }}
     return cluster.Namespaces[namespaceName].State == effectiveaccessscope.Included
```

Comment:
the Namespaces map lookup could return `nil` on cache miss. Then `.State` would cause a panic.

Reply:
I removed this function as we can use sub scope checker